pageserver/src/tenant/delete.rs

```diff
@@ -0,0 +1,569 @@
+use std::sync::Arc;
+
+use anyhow::Context;
+use pageserver_api::models::TimelineState;
+use tracing::{debug, error, info, instrument, warn, Instrument, Span};
+use utils::{
+    crashsafe, fs_ext,
+    id::{TenantId, TimelineId},
+};
+
+use crate::{
+    config::PageServerConf,
+    task_mgr::{self, TaskKind},
+    tenant::{remote_timeline_client, DeleteTimelineError},
+    InitializationOrder,
+};
+
+use super::{
+    metadata::TimelineMetadata,
+    remote_timeline_client::{PersistIndexPartWithDeletedFlagError, RemoteTimelineClient},
+    DeletionGuard, Tenant, Timeline,
```

Comment:
DeletionGuard should be moved in as well, could go to the bottom as it's not really that interesting right?

Reply:
Fixed https://github.com/neondatabase/neon/pull/4619/commits/51691e03c02416977c4521b2343f009556e6d68e